src/pages/SearchPage.js

```diff
@@ -35,212 +32,190 @@ const propTypes = {
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [searchValue]);
 
     /**
      * Returns the sections needed for the OptionsSelector
      *
      * @returns {Array}
      */
-    getSections() {
+    const getSections = () => {
         const sections = [];
         let indexOffset = 0;
 
-        if (this.state.recentReports.length > 0) {
+        if (searchOptions.recentReports.length > 0) {
             sections.push({
-                data: this.state.recentReports,
+                data: searchOptions.recentReports,
                 shouldShow: true,
                 indexOffset,
             });
-            indexOffset += this.state.recentReports.length;
+            indexOffset += searchOptions.recentReports.length;
         }
 
-        if (this.state.personalDetails.length > 0) {
+        if (searchOptions.personalDetails.length > 0) {
             sections.push({
-                data: this.state.personalDetails,
+                data: searchOptions.personalDetails,
                 shouldShow: true,
                 indexOffset,
             });
-            indexOffset += this.state.recentReports.length;
+            indexOffset += searchOptions.recentReports.length;
         }
 
-        if (this.state.userToInvite) {
+        if (searchOptions.userToInvite) {
             sections.push({
-                data: [this.state.userToInvite],
+                data: [searchOptions.userToInvite],
                 shouldShow: true,
                 indexOffset,
             });
         }
 
         return sections;
-    }
+    };
 
-    searchRendered() {
+    const searchRendered = () => {
         Timing.end(CONST.TIMING.SEARCH_RENDER);
         Performance.markEnd(CONST.TIMING.SEARCH_RENDER);
-    }
-
-    updateOptions() {
-        const {recentReports, personalDetails, userToInvite} = OptionsListUtils.getSearchOptions(
-            this.props.reports,
-            this.props.personalDetails,
-            this.state.searchValue.trim(),
-            this.props.betas,
-        );
-        this.setState({
-            userToInvite,
-            recentReports,
-            personalDetails,
-        });
-    }
+    };
+
+    const onChangeText = (value = '') => {
+        setSearchValue(value);
+    };
 
     /**
      * Reset the search value and redirect to the selected report
      *
      * @param {Object} option
      */
-    selectReport(option) {
+    const selectReport = (option) => {
         if (!option) {
             return;
         }
-
         if (option.reportID) {
-            this.setState(
-                {
-                    searchValue: '',
-                },
-                () => {
-                    Navigation.dismissModal(option.reportID);
-                },
-            );
+            setSearchValue('');
+            Navigation.dismissModal(option.reportID);
```

Comment:
Please add this as a callback to to the `setSearchValue` like we did previously.

Reply:
we can do it like this with setSearchValue from the useState hook and I don't really see the reason why it should be run as a callback

Reply:
@Piotrfj Can you please address this comment?

Reply:
@allroundexperts we cannot do it the same way we did in the class component. And I don't really see the necessity to do such here. What's more: the component always unmounts when we navigate to the report page, so the next time we open search page the search value is an empty string - no need to clear the input then.